slides/aml-10-021918-trees-forests/index.html

```diff
@@ -0,0 +1,724 @@
+.center[
+![:scale 70%](images/img_1.png)
+]
+???
+
+---
+class: split-40
+# Building Trees
+
+.column[
+![:scale 70%](images/img_2.png)
+ - Continuous features:
+ - “questions” are thresholds on single features.
+ - [Other methods are possible but not as common]
+ - For each split: exhaustive search over all features and thresholds!
+ - Minimize “impurity”
+]
+.column[
+![:scale 50%](images/img_3.png)
+---
+![:scale 50%](images/img_4.png)
+]
+
+???
+Does the layout look okay??
+---
+
+# Criteria (for classification)
+ - Gini Index: 
+   --
+   .center[![:scale 50%](images/img_5.png)]
+ - Cross-Entropy:
+   --
+   .center[![:scale 50%](images/img_6.png)]
+
+
+ 
+ ![:scale 5%](images/img_7.png) observations in node m
+ 
+ ![:scale 3%](images/img_8.png) classes
+ 
+ ![:scale 5%](images/img_9.png) distribution over classes in node m
+???
+---
+
+# Prediction
+
+.center[
+![:scale 80%](images/img_10.png)
+]
+
+ - Traverse tree based on feature tests
+ - Predict most common class in leaf
+
+???
+
+---
+
+# Regression trees
+
+ - Impurity Criteria:
+ - Mean Squared Error
+ - Mean Absolute Error
+ 
+ 
+ - Prediction:
+ - Predict mean.
+ 
+
+ - Without regularization / pruning:
+ - Each leaf often contains a single point to be “pure”
+???
+
+---
+
+# Visualizing trees with sklearn
+
+.smaller[
+```python
+from sklearn.datasets import load_breast_cancer
+cancer = load_breast_cancer()
+```
+```python
+X_train, X_test, y_train, y_test = train_test_split(cancer.data,canc
+er.target, stratify=cancer.target,random_state=0)
+```
+```python
+#tree visualization
+```
+```python
+from sklearn.tree import DecisionTreeClassifier, export_graphviz
+tree = DecisionTreeClassifier(max_depth=2)
+tree.fit(X_train, y_train) 
+```
+```python
+tree_dot = export_graphviz(tree, out_file=None, feature_names=cancer.feature_names)
+print(tree_dot)
+```
+]
+
+???
+---
+# Visualizing trees with sklearn
+.center[
+![:scale 80%](images/img_11.png)
+]
+???
+---
+
+
+# Showing dot files in Jupyter
+ - First install graphvis C library:
+   conda install graphviz
+ - Then install graphviz python library:
+ - pip install graphviz
+
+ .center[
+ ![:scale 60%](images/img_12.png)
+ ]
+
+
+???
+---
+class:spacious
+
+# Parameter Tuning
+
+ - Pre-pruning and post-pruning (not in sklearn yet)
+ - Limit tree size (pick one):
+   max_depth
+   --
+   max_leaf_nodes
+   --
+   min_samples_split
+   --
+   (and more)
+???
+---
+class:spacious
+
+# No pruning
+.center[
+![:scale 100%](images/img_13.png)
+]
+???
+
+---
+
+# max_depth = 4
+
+.center[
+![:scale 100%](images/img_14.png)
+]
+
+???
+---
+
+# max_leaf_nodes = 8
+.center[
+![:scale 50%](images/img_15.png)
+]
+
+???
+---
+
+# min_samples_split = 50
+.center[
+![:scale 70%](images/img_16.png)
+]
+
+???
+---
+
+.smaller[
+```python
+from sklearn.model_selection import GridSearchCV
+param_grid = {'max_depth':range(1, 7)}
+grid = GridSearchCV(DecisionTreeClassifier(random_state=0),param_grid=param_grid,
+cv=10)
```

Comment:
please use pep8 indentation. In this case, line up with DecisionTree.

Reply:
Resolved